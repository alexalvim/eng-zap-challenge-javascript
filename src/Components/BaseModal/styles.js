import styled from 'styled-components';

import Colors from '../../colors';
import Spaces from '../../spaces';

export const OverlayWrapper = styled.div`
  align-items: center;
  background-color: ${Colors.overlayColor}
  display: flex;
  height: 100%;
  justify-content: center;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 5;
`;

export const ModalContent = styled.div`
  background-color: ${Colors.lighterColor};
  border-radius: 5px;
  height: 100%;
  position: relative;
  width: 100%;

  @media(min-width: ${Spaces.tabletBreakpoint}) {
    height: auto;
    max-height: 600px;
    max-width: 500px;
  }
`;

export const CloseButton = styled.button`
  background-color: ${Colors.lighterColor};
  border: none;
  border-radius: 50%;
  box-shadow: 1px 3px 3px rgba(0,0,0,0.5);
  cursor: pointer;
  height: 30px;
  padding: ${Spaces.half};
  position: absolute;
  right: ${Spaces.base};
  top: ${Spaces.base};
  width: 30px;
  z-index: 1;
`;

export const closeButtonIconStyles = {
  color: Colors.darkerColor
}