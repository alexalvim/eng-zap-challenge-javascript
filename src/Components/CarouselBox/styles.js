import styled from 'styled-components';

import Spaces from '../../spaces';
import Typo from '../../typo';

export const ContentWrapper = styled.div`
  border-radius: 5px;
  box-shadow: 0px 3px 5px rgba(0,0,0,0.2);
  display: flex;
  flex-direction: column;
  margin: ${Spaces.double} 0;
  overflow: hidden;

  @media(min-width: ${Spaces.tabletBreakpoint}){
    flex-direction: row;
  }
`;

export const CarouselContainer = styled.div`
  width: 100%;

  .slick-arrow:before {
    opacity: 1;
    text-shadow: 1px 3px 3px rgba(0,0,0,0.5);
  }

  .slick-arrow.slick-disabled:before {
    opacity: .5;
  }

  .slick-next {
    right: 5px;
    z-index: 1;
  }

  .slick-prev {
    left: 5px;
    z-index: 1;
  }

  @media(min-width: ${Spaces.tabletBreakpoint}){
    width: 350px;
  }
`;

export const ImageContainer = styled.div`
  align-items: center;
  display: flex !important;
  height: 300px;
  justify-content: center;
  overflow: hidden;
  position: relative;

  > img {
    width: auto;
    height: 300px;
    position: absolute;
    left: 50%;
    transform: translateX(-50%);
  }

  @media(min-width: ${Spaces.tabletBreakpoint}){
    max-width: 350px;
  }
`;

export const TextHolder = styled.div`
  display: flex;
  flex-direction: column;
  padding: ${Spaces.base};
`;

export const titleWrapperStyles = `
  font-size: ${Typo.large};
  margin-bottom: ${Spaces.base};
`;