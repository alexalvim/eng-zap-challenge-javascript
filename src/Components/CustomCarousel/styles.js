import styled from 'styled-components';

export const CarouselContainer = styled.div`
  width: 100%;

  .slick-arrow:before {
    opacity: 1;
    text-shadow: 1px 3px 3px rgba(0,0,0,0.5);
  }

  .slick-arrow.slick-disabled:before {
    opacity: .5;
  }

  .slick-next {
    right: 5px;
    z-index: 1;
  }

  .slick-prev {
    left: 5px;
    z-index: 1;
  }
`;

export const ImageContainer = styled.div`
  align-items: center;
  display: flex !important;
  justify-content: center;
  overflow: hidden;
  position: relative;

  > img {
    width: auto;
    position: absolute;
    left: 50%;
    transform: translateX(-50%);
  }
  ${({height}) => height &&
  `
    height: ${height}px;

    > img {
      height: ${height}px;
    }
  `}
`;