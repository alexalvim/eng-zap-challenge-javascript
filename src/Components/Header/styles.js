import styled from 'styled-components';

import Colors from '../../colors';
import Spaces from '../../spaces';

export const ContentWrapper = styled.div`
  background-color: ${Colors.lighterColor};
  color: ${Colors.lighterColor};
  padding: ${Spaces.double};
  width: 100%;

  ${({activePortal}) => activePortal && `border-bottom: solid 2px ${Colors.portalMainColor[activePortal]};`}
`;

export const Container = styled.div`
  margin: 0 auto;
  max-width: ${Spaces.containerMaxWidth};
  width: 100%;
`;

export const Logo = styled.img`
  height: 40px;
`;