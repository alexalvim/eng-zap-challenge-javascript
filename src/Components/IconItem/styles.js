import styled from 'styled-components';

import Colors from '../../colors';
import Typo from '../../typo'
import Spaces from '../../spaces';

export const ContentWrapper = styled.div`
  align-items: center;
  display: flex;
`;

export const Label = styled.span`
  color: ${Colors.darkerColor};
  font-size: ${Typo.medium};
`;

export const iconStyles = {
  marginRight: Spaces.base,
}