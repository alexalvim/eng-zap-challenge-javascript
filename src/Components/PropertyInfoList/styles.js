import styled from 'styled-components';

import Spaces from '../../spaces';

export const List = styled.ul`
  > li {
    margin-bottom: ${Spaces.base};
  }
`;