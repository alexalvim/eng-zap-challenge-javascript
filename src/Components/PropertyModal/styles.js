import styled from 'styled-components';

import Spaces from '../../spaces';
import Typo from '../../typo';
import Colors from '../../colors';

export const TextWrapper = styled.div`
  padding: ${Spaces.double};
`;

export const Title = styled.h2`
  font-size: ${Typo.large};
  margin-bottom: ${Spaces.double};

  ${({activePortal}) => activePortal && `color: ${Colors.portalMainColor[activePortal]}`}
`;

export const Price = styled.span`
  color: ${Colors.darkerColor};
  display: inline-block;
  font-size: ${Typo.medium};
  margin-top: ${Spaces.double};
`;