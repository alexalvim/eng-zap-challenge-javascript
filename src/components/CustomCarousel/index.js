import React from 'react';
import Slider from 'react-slick';

import {
  CarouselContainer,
  ImageContainer
} from './styles'

export default ({ images, height, alt }) => {
  const settings = {
    infinite: false,
    speed: 500,
    slidesToShow: 1,
    slidesToScroll: 1
  };
  return (
    <CarouselContainer>
      <Slider {...settings}>
        {images.map((image, i) => (
          <ImageContainer height={height}  key={i}>
            <img src={image} alt={alt}/>
          </ImageContainer>
        ))}
      </Slider>
    </CarouselContainer>
   );
}