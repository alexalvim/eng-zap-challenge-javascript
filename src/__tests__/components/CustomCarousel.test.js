import React from 'react';
import { shallow, mount } from 'enzyme';
import 'jest-styled-components'

import CustomCarousel from '../../components/CustomCarousel';
import { ImageContainer } from '../../components/CustomCarousel/styles';

describe('Testing CustomCarousel', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <CustomCarousel
        images={['url']}
        alt='altImage'
        height={10}/>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should pass correct height to ImageContainer', () => {
    const wrapper = mount(
      <CustomCarousel
        images={['url']}
        alt='altImage'
        height={10}/>
    );

    expect(wrapper.find(ImageContainer).prop('height')).toEqual(10);
  });

  it('should use correct alt', () => {
    const wrapper = mount(
      <CustomCarousel
        images={['url']}
        alt='altImage'
        height={10}/>
    );

    expect(wrapper.find('img').prop('alt')).toEqual('altImage');
  });
});
