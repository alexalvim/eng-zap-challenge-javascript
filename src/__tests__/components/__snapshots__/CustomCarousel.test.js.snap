// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CustomCarousel should render correctly 1`] = `ShallowWrapper {}`;
